---
<div style="text-align: center;" >
    <h3>
        授業名 : データサイエンスプログラミング演習Ⅰ
    </h3>
    <h1>
        <b>レポート : スポーツジムにおける会員データ分析</b>
    </h1>
</div>
<div style="text-align: left">
    学籍番号 : 1922076<br/>
    氏名 : 南 昂汰
</div>

---

1 . 全体の会員データから読み取れることを調査する。例えば、どの会員区分やキャンペーン区分の会員が多いか。

In [2]:
import pandas as pd
ca_g=pd.read_csv("campaign_gym.csv",encoding="UTF-8")
cl_g=pd.read_csv("class_gym.csv",encoding="UTF-8")
u_l=pd.read_csv("use_log.csv",encoding="UTF-8")
cu_g=pd.read_csv("customer_gym.csv",encoding="UTF-8")
cu_g["customer_id"]=cu_g["customer_id"].str.upper()
joined=pd.merge(cu_g,cl_g,on="class",how="left")
joined=pd.merge(joined,ca_g,on="campaign_id",how="left")

In [3]:
joined.groupby("class_name").count()["customer_id"]

class_name
オールタイム    2045
デイタイム     1019
ナイト       1128
Name: customer_id, dtype: int64

Out[3]より、オールタイムの契約が最も多い2045人であることが分かる。これはデイタイムの契約数1019人の約2倍である。

<br>
<br>
<br>
<br>
<br>

In [4]:
joined.groupby("campaign_name").count()["customer_id"]

campaign_name
入会費半額     650
入会費無料     492
通常       3050
Name: customer_id, dtype: int64

In [5]:
print(len(joined[joined["campaign_name"].isin(["入会費半額","入会費無料"])])/len(joined)*100)

27.24236641221374


Out[4]と[5]より、キャンペーンで入会した人は全体の27%であり、無料で入会した人数が最も少ないことが分かる。

In [6]:
joined.groupby("gender").count()["customer_id"]

gender
F    1983
M    2209
Name: customer_id, dtype: int64

Out[6]より、性別にあまり大差はないが、男性のほうが多いことが分かる。

In [7]:
joined["start_date"]=pd.to_datetime(joined["start_date"])
joined["start_year"]=joined["start_date"].dt.strftime("%Y")
joined.groupby("start_year").count()["customer_id"]

start_year
2015     492
2016     781
2017    1223
2018    1467
2019     229
Name: customer_id, dtype: int64

Out[7]より、2018年に入会した人が最も多く、入会人数は年を追うごとに増加していることが分かる。

In [8]:
joined.groupby("is_deleted").count()["customer_id"]

is_deleted
0    2842
1    1350
Name: customer_id, dtype: int64

In [9]:
print(len(joined[joined["is_deleted"].isin([1])])/len(joined)*100)

32.20419847328244


Out[8]と[9]より、会員の32%が退会していることが分かる。

In [10]:
joined["start_date"]=pd.to_datetime(joined["start_date"])
joined["start_month"]=joined["start_date"].dt.strftime("%m")
joined.groupby("start_month").count()["customer_id"]

start_month
01    343
02    300
03    286
04    331
05    416
06    406
07    357
08    354
09    349
10    289
11    301
12    460
Name: customer_id, dtype: int64

Out[10]より、3月に入会した人数が286人と最も少なく、12月が最も多い460人であることが分かる。

<br>
<br>
<br>
<br>
<br>
<br>
<br>
<br>
<br>
<br>

<br>
<br>

2 . 最新の会員データから読み取れることを調査する。例えば、どの会員区分やキャンペーン区分の会員が多いか、全体の会員データとの違いなどを考察する。

In [11]:
joined["end_date"]=pd.to_datetime(joined["end_date"])
newer=joined.loc[(joined["end_date"]>=pd.to_datetime("20190401"))|(joined["end_date"].isna())]

In [12]:
newer.groupby("class_name").count()["customer_id"]

class_name
オールタイム    1381
デイタイム      675
ナイト        786
Name: customer_id, dtype: int64

全体のオールタイムの契約数が2045人ということが1ページ目のOut[3]より判明したが、そのうちの1381人が最新の会員であることがOut[12]より分かる。これは全体のデータのうち、**「67.5%」**である。<br>
全体のデイタイムの契約数が1019人ということが1ページ目のOut[3]より判明したが、そのうちの675人が最新の会員であることがOut[12]より分かる。これは全体のデータのうち、**「66.2%」**である。<br>
全体のナイトの契約数が1128人ということが1ページ目のOut[3]より判明したが、そのうちの786人が最新の会員であることがOut[12]より分かる。これは全体のデータのうち、**「69.6%」**である。

<br>
<br>

3 . 利用履歴から読み取れること、退会しやすい会員や継続しやすい会員の特徴などを分析する。

さきほどのデータから分かったをまとめると、<br>
最新のデイタイム会員が全体のデイタイム会員に占める割合：66.2%<br>
最新のオールタイム会員が全体のオールタイム会員に占める割合：67.5%<br>
最新のナイト会員が全体のナイト会員に占める割合：69.6%<br>
<br>
つまり、ナイト会員が継続しやすく、デイタイム会員が退会しやすいという傾向があると言える。